educe/learning/edu_input_format.py

```diff
@@ -28,13 +31,16 @@ def _dump_edu_input_file(docs, f):
             edu_txt = edu.text().replace('\n', ' ')
             # subgroup: sentence identifier, backoff on EDU id
             sent_idx = edu2sent[i]
-            subgroup = ('{}_sent{}'.format(grouping, sent_idx)
-                        if sent_idx is not None
-                        else edu_gid)
+            if sent_idx is None:
+                subgroup = edu_gid
+            elif isinstance(sent_idx, six.string_types):
```

Comment:
Are there cases where we really do want to store strings in `edu2sent` ?
What I had in mind with such arrays was to turn them into numpy arrays or better, masked arrays (to store missing or invalid data, which is why we currently have None in  ̀edu2sent`).
Storing strings in `edu2sent` would defeat this purpose, so unless you have really compelling reasons to diverge, I would prefer you revert this precise change.